File: ios/CustomKeyboard/CustomKeyboard.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CustomKeyboard, NSObject)

RCT_EXTERN_METHOD(isEnabled:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isActivated:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/CustomKeyboard/CustomKeyboard.swift
@objc(CustomKeyboard)
class CustomKeyboard: NSObject {
    @objc func isEnabled(_ resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock){
        resolve(false)
    }

    @objc func isActivated(_ resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock){
        resolve(false)
    }
    
    func log(for text: String) {
        guard text.count > 0 else { return }
        NSLog("\(text)")
    }
    
    func Foo(_ format: String, _ args: CVarArg...) {
        NSLog(format, args)
    }
}
